src/components/internals/create-select-styles.js

```diff
@@ -157,22 +157,27 @@ const multiValueRemoveStyles = () => base => ({
   },
 });
 
-const groupStyles = () => base => ({
+const groupStyles = props => base => ({
   ...base,
   padding: 0,
-  '&:not(:firstChild)': {
-    borderTop: `1px ${vars.borderColorSeparator} solid`,
+  '&:not(:first-child)': {
+    borderTop: props.shouldDivideOptionGroups
+      ? `1px solid ${vars.colorGray}`
+      : base.borderTop,
   },
 });
 
-const groupHeadingStyles = () => base => ({
+export const groupHeadingStyles = () => base => ({
```

Comment:
Why this change?

Reply:
this was something I added while figuring out the implementation and forgot to remove. Will get rid of this